app/models/bishop.rb

```diff
@@ -0,0 +1,10 @@
+class Bishop < Piece
+  def valid_move?(x_destination, y_destination)
+    return false unless super(x_destination, y_destination)
+    return false if obstructed?(x_destination, y_destination)
```

Comment:
Maybe move obstructed to superclass?

Knight is the only piece that can't be obstructed. You could change the super method signature to:
```
valid_move?(x_destination, y_destination, obstruct=true)
```

Then just `super(x, y, false)` in the Knight. That will help a bit with the DRY.